batch/batch/front_end/templates/table_search.html

```diff
@@ -0,0 +1,67 @@
+{% macro table_search(component_name, endpoint) %}
+  <style>
+    #{{ component_name }} {
+      margin: 10px;
+      margin-left: 0px;
+    }
+    #{{ component_name }}-inner {
+      height: 15rem;
+      width: 65vw;
+      display: flex;
+      align-items: center;
+    }
+    #{{ component_name }}-input-box {
+      background-color: #efefef;
+      padding: 0.75rem;
+      border-radius: 3px;
+      font-size: 1rem;
+      resize: none;
+      width: 450px;
+      white-space: pre;
+      height: 10rem;
+    }
+    #{{ component_name}}-buttons {
+      margin-left: 1rem;
+      height: 100%;
+      display: flex;
+      flex-direction: column;
+      justify-content: space-between;
+    }
+    #{{ component_name }}-submit {
+      margin-bottom: 0.9rem;
+    }
+    #{{ component_name }}-submit:hover {
+      background-color: #283870;
+      color: #fff;
+    }
+    #{{ component_name }}-info {
+      margin-top: 0.4rem;
+      color: #8398e0;
+      font-size: 2rem;
+    }
+    #{{ component_name }}-info:hover {
+      cursor: pointer;
+      color: #283870;
+    }
+  </style>
+  <div class="flex-col">
+    <div id="{{ component_name }}" class="flex-col-align-right">
+      <form method="GET" action="{{ endpoint }}">
+        <div id="{{ component_name }}-inner">
+          <div>
+            <textarea id="{{ component_name }}-input-box" autofocus name="q">{% if q %}{{ q }}{% endif %}</textarea>
```

Comment:
I'm pretty sure we don't need a CSRF element in the form since it's a GET, but @danking can you confirm?

```
<input type="hidden" name="_csrf" value="{{ csrf_token }}"/>
```

Reply:
I was being silly. We don't need this.

Reply:
Yeah, CSRF is to protect the user from a malicious website that convinces them to take an action at hail.is. This is only bad if the action is bad. For example, submitting a batch could cost money. Viewing the batches page should generally be safe b/c if the user unexpectedly sees a list of their batches, that's weird, but doesn't cost money, delete data, etc.